debug_module.tex

```diff
@@ -52,18 +52,20 @@ \section{Debug Module Interface (DMI)} \label{dmi}
 \section{Reset Control} \label{reset}
 
 The Debug Module controls a global reset signal, which can
-reset, or hold in reset, every component in the platform,
+reset, or hold in reset, every component in the platform
 except for the Debug Module and Debug
 Transport Modules. The purpose of this feature is to allow debugging
 programs from the first instruction executed, so exactly what is affected
 by this reset is implementation dependent. The Debug Module's own state and registers are
-generally reset at power-up and if and only if
+generally reset at power-up if and only if
 \Fdmactive in \Rdmcontrol is 0. This means that the halt state of all harts is
 maintained provided that \Fdmactive is 1, although trigger CSRs may be cleared.
 
 Due to clock and power domain crossing issues,
 it may not be possible to perform arbitrary DMI accesses across reset,
-even if the Debug Module state is maintained. It must always be possible to access
+even if the Debug Module state is maintained. The debugger must not perform
+a reset while it's expecting a DMI access to complete. It must always be
```

Comment:
So, in rocket-chip, you can't actually get across the CDC boundary while reset is asserted, because the clocks aren't running on the DM side (even though the DM values are maintained, your DMI access can't complete). It doesn't matter what the source of reset is (could be ndmreset, or me holding the reset button down).

DMCONTROL is on the JTCK domain so it's fine to read that one (which is necessary so that you can actually clear ndmreset).

All of this is very implementation specific, but I'm not sure how you would get around it, so wanted to specify the minimum that the implementation has to support.